src/app/modules/shared/components/tables/mat-fsp-table/mat-fsp-table.component.html

```diff
@@ -0,0 +1,40 @@
+ <!--<app-edit-supplier #editsupplier></app-edit-supplier>
+ <button mat-flat-button #thisbtn>Hello</button>-->
+  <div class="mat-elevation-z8">
+    <mat-form-field>
+      <input matInput placeholder="Filter" #input>
+    </mat-form-field>
+
+    <table mat-table [dataSource]="dataSource" matSort>
+
+      <ng-container *ngFor="let colDef of columnDefinitions; let i = index" [matColumnDef] = "colDef">
+        <th mat-header-cell *matHeaderCellDef mat-sort-header> {{columnNames[i]}} </th>
+        <td mat-cell *matCellDef="let row"> {{row[colDef]}} </td>
+      </ng-container>
+      
+      <ng-container matColumnDef = "buttonsColumn">
+        <th mat-header-cell *matHeaderCellDef> ДіЇ </th>
+        <td mat-cell *matCellDef="let row"> 
+          <!--<ng-container *ngFor="let button of buttons">
+            <button [widget]="button"></button>
+          </ng-container>-->
+          
+          <button mat-flat-button *ngIf="info.length > 0">Info</button>
```

Comment:
regarding the previous comment: try to change isVisible bound property on buttons click